src/otx/core/model/module/instance_segmentation.py

```diff
@@ -0,0 +1,152 @@
+        self.val_metric.reset()
+
+    def on_test_epoch_start(self) -> None:
+        """Callback triggered when the test epoch starts."""
+        self.test_metric.reset()
+
+    def on_validation_epoch_end(self) -> None:
+        """Callback triggered when the validation epoch ends."""
+        self._log_metrics(self.val_metric, "val")
+        self.val_metric.reset()
+
+    def on_test_epoch_end(self) -> None:
+        """Callback triggered when the test epoch ends."""
+        self._log_metrics(self.test_metric, "test")
+        self.test_metric.reset()
+
+    def _log_metrics(self, meter: MeanAveragePrecision, subset_name: str) -> None:
+        results = meter.compute()
+        for metric, value in results.items():
+            if not isinstance(value, Tensor):
+                log.debug("Cannot log item which is not Tensor")
+                continue
+            if value.numel() != 1:
+                log.debug("Cannot log Tensor which is not scalar")
+                continue
+
+            self.log(
+                f"{subset_name}/{metric}",
+                value,
+                sync_dist=True,
+                prog_bar=True,
+            )
+
+    def validation_step(self, inputs: InstanceSegBatchDataEntity, batch_idx: int) -> None:
+        """Perform a single validation step on a batch of data from the validation set.
+
+        Args:
+            inputs (InstanceSegBatchDataEntity): The input data for the validation step.
+            batch_idx (int): The index of the current batch.
+
+        Raises:
+            TypeError: If the predictions are not of type InstanceSegBatchPredEntity.
+
+        Returns:
+            None
+        """
+        preds = self.model(inputs)
+
+        if not isinstance(preds, InstanceSegBatchPredEntity):
+            raise TypeError(preds)
+
+        self.val_metric.update(
+            **self._convert_pred_entity_to_compute_metric(preds, inputs),
+        )
+
+    def _convert_pred_entity_to_compute_metric(
+        self,
+        preds: InstanceSegBatchPredEntity,
+        inputs: InstanceSegBatchDataEntity,
+    ) -> dict[str, list[dict[str, Tensor]]]:
+        """Convert the prediction entity to the format required by the compute metric function."""
+        pred_info = []
+        target_info = []
+
+        for bboxes, masks, scores, labels in zip(
+            preds.bboxes, preds.masks, preds.scores, preds.labels,
+        ):
+            pred_info.append({
+                "boxes": bboxes.data,
+                "masks": masks.data,
+                "scores": scores,
+                "labels": labels,
+            })
+
+        for imgs_info, bboxes, masks, polygons, labels in zip(
+            inputs.imgs_info,
+            inputs.bboxes,
+            inputs.masks,
+            inputs.polygons,
+            inputs.labels,
+        ):
+            bit_masks = masks if len(masks) else polygon_to_bitmap(polygons, *imgs_info.ori_shape)
```

Comment:
We cannot execute `polygon_to_bitmap` in the main thread (training loop). This is a time-consuming job and should be done in the data preparation threads (not main thread) in parallel and asynchronously. Please do this at the data preparation stage.

Reply:
As highlighted in the previous comment, storing bitmap tensors in CUDA memory incurs a considerable expense. Therefore, I've opted for polygons as an alternative. However, there's a limitation as torchmetric specifically accepts bitmap masks, necessitating the conversion of polygons to bitmap masks. It's worth noting that converting batch ground truth polygons consumes less memory than caching batch ground truth masks but might potentially slow down the validation process.

If we decide to handle this conversion in `PackDetInputs`, it introduces the use of BitmapMasks, requiring a series of conversions between CPU and GPU (CPU -> GPU -> CPU -> GPU). The dilemma lies in determining the optimal approach, considering the trade-offs between memory usage and processing speed. I'm uncertain which path is more favourable in this scenario.